docs_app/content/guide2/writing-readable-rxjs.md

```diff
@@ -0,0 +1,251 @@
+# ????. Writing Readable RxJS
+
+# TLDR:
+
+- Don't write "mega observables". Break them up into smaller observables and functions.
+- Use well-named custom operators.
+- Add comments about the "whys", not the "hows".
+- Be sure to have your code reviewed thoroughly.
+
+## Readability vs Understanding
+
+Often, whether it's RxJS-related, or JavaScript, or SQL, or any programming language, when people say "This code isn't readable!", what they often mean is "I don't understand what this code does". The problem at 
+hand is that one developer used some abstractions provided by a language or library, and, at least at the time, understood details about what was happening "under the hood" of these abstractions. They understood not just "how" it was working, but _why_ they needed it to work that way. Then another developer comes along, and they "can't read" the first developers code. The fact is, they can indeed "read" the code, but they don't _understand_ the code. Why does the next developer not understand the code?
+
+The next developer will have a hard time **understanding** your code if they lack one or both of these things:
+
+1. Context around why the code is doing what it is doing.
+2. Knowledge of what the abstractions your using do.
+
+It's very important, in _all_ of your code, not just the RxJS-related stuff, that these two things are conveyed cleanly, and concisely, to the next developer to come across your code.
+
+## Break Your Code Up
+
+You wouldn't write a "mega function" in most cases, don't write "mega observables". Observables are effectively functions. As functions don't do anything until you call the, observables don't do anything until you _subscribe_ to them. When you're building an observable by chaining operators together, what you're doing is effectively writing a function by adding lines of code to it, so to speak. Very long functions can be hard to understand because they can be hard to follow. Variable declarations can slowly move away from where they are used over the evolution of the code, and the local variable context gets cluttered with names that can start to conflict with one another.
+
+The very nature of reactive programming with observables is that each piece is processed as an individual "step". Every operator returns a new observable. This means that you can break single "mega observable" into smaller observables. This can help with unit testing, in particular.
```

Comment:
Consider changing the last sentence to: "This can help with understanding and with unit testing" (Since this section has been stressing understand-ability.)